service_openrc_linux.go

```diff
@@ -166,19 +165,28 @@ func (s *openrc) Run() (err error) {
 }
 
 func (s *openrc) Status() (Status, error) {
+	// rc-service uses the errno library for its exit codes:
+	// errno 0 = service started
+	// errno 1 = EPERM 1 Operation not permitted
+	// errno 2 = ENOENT 2 No such file or directory
+	// errno 3 = ESRCH 3 No such process
+	// for more info, see https://man7.org/linux/man-pages/man3/errno.3.html
 	_, out, err := runWithOutput("rc-service", s.Name, "status")
 	if err != nil {
-		return StatusUnknown, err
-	}
-
-	switch {
-	case strings.HasPrefix(out, "Running"):
-		return StatusRunning, nil
-	case strings.HasPrefix(out, "Stopped"):
-		return StatusStopped, nil
-	default:
-		return StatusUnknown, ErrNotInstalled
+		exitCode := err.(*exec.ExitError).ExitCode()
```

Comment:
Sorry, let's do one more thing. When we do this type assert, use the comma, ok notation. If not Exit Error, then return StatusUnknown Err. then only call ExitCode if it is that type.

Reply:
I added another commit to address your requested changes. Hope this covers what you needed.